core/src/main/java/org/apache/iceberg/SystemConfigs.java

```diff
@@ -53,6 +53,17 @@ private SystemConfigs() {}
           Math.max(2, Runtime.getRuntime().availableProcessors()),
           Integer::parseUnsignedInt);
 
+  /**
+   * The size of the queue in ParallelIterable. This queue limits the memory usage of manifest
+   * reader.
+   */
+  public static final ConfigEntry<Integer> ITERATOR_QUEUE_SIZE =
+      new ConfigEntry<>(
+          "iceberg.iterator.queue-size",
```

Comment:
the config naming implies (at least to me) that this is being applied to all iterators being used in Iceberg, which isn't the case